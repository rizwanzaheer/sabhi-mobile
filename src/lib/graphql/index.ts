import { Agent } from 'daf-core'
import AsyncStorage from '@react-native-community/async-storage'

export interface Context {
  agent: Agent
}

const isSelected = async (identity: any, args: any, ctx: Context) => {
  const did = await AsyncStorage.getItem('selectedIdentity')
  return identity.did === did
}

const setViewer = async (_: any, args: { did: string }, ctx: Context) => {
  return await AsyncStorage.setItem('selectedDid', args.did)
}

const request = async (_: any, args: any, ctx: Context) => {
  const did = await AsyncStorage.getItem('selectedIdentity')

  if (did !== null) {
    return {
      did,
      __typename: 'Identity',
    }
  }
}

const credentialView = async (_: any, args: any, ctx: Context) => {
  const credential = args.credential
  return {
    credential,
    __typename: 'Credential',
  }
}

const viewer = async (_: any, args: any, ctx: Context) => {
  const did = await AsyncStorage.getItem('selectedIdentity')

  if (did !== null) {
    return {
      did,
      __typename: 'Identity',
    }
  } else {
    return {
      did: null,
    }
  }
}

export const resolvers = {
  Identity: {
    isSelected,
  },
  Query: {
    viewer,
  },
  Mutation: {
    setViewer,
  },
  Message: {
    viewer,
  },
}

export const typeDefs = `
  extend type Identity {
    isSelected: Boolean
  }
  extend type Message {
    viewer: Identity
  }
  extend type Query {
    viewer: Identity
  }
  extend type Mutation {
    setViewer(did: String): Boolean
  }
`
