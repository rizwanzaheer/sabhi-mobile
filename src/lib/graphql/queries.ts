import gql from 'graphql-tag'

export const GET_SECRET_KEY = gql`
  query managedIdentitySecret($did: String, $type: String) {
    managedIdentitySecret(did: $did, type: $type)
  }
`

export const GET_IDENTITY = gql`
  query getIdentity($did: ID!) {
    identity(did: $did) {
      isManaged
      did
      shortId: shortDid
      shortDid
      firstName: latestClaimValue(type: "firstName")
      lastName: latestClaimValue(type: "lastName")
      profileImage: latestClaimValue(type: "profileImage")
    }
  }
`

export const GET_ALL_IDENTITIES = gql`
  query GetAllIdentities {
    identities {
      isManaged
      isSelected
      did
      shortId: shortDid
      shortDid
      firstName: latestClaimValue(type: "firstName")
      lastName: latestClaimValue(type: "lastName")
      profileImage: latestClaimValue(type: "profileImage")
    }
  }
`

export const GET_CREDENTIALS_FOR_IDENTITY = gql`
  fragment ShortProfile on Identity {
    did
    shortId: shortDid
    shortDid
    firstName: latestClaimValue(type: "firstName")
    lastName: latestClaimValue(type: "lastName")
    profileImage: latestClaimValue(type: "profileImage")
  }

  query Credentials($selectedIdentity: ID!) {
    identity(did: $selectedIdentity) {
      isSelected
      ...ShortProfile
      receivedCredentials: credentials(
        input: { subject: [$selectedIdentity] }
      ) {
        hash
        issuer {
          ...ShortProfile
        }
        subject {
          ...ShortProfile
        }
        credentialSubject
        claims {
          type
          value
          isObj
        }
      }
    }
  }
`

export const GET_VIEWER = gql`
  query getViewer {
    viewer {
      did
      shortId: shortDid
      shortDid
      profileImage: latestClaimValue(type: "profileImage")
    }
  }
`

export const GET_VIEWER_CREDENTIALS = gql`
  fragment ShortProfile on Identity {
    did
    shortId: shortDid
    shortDid
    firstName: latestClaimValue(type: "firstName")
    lastName: latestClaimValue(type: "lastName")
    profileImage: latestClaimValue(type: "profileImage")
  }
  query getViewer($selectedIdentity: ID!) {
    viewer {
      ...ShortProfile
      isManaged
      isSelected
    }
    credentials(input: { subject: [$selectedIdentity] }) {
      hash
      issuer {
        ...ShortProfile
      }
      subject {
        ...ShortProfile
      }
      credentialSubject
      claims {
        type
        value
        isObj
      }
    }
  }
`

export const SET_VIEWER = gql`
  mutation setViewer($did: String!) {
    setViewer(did: $did) {
      did
    }
  }
`

export const IMPORT_IDENTITY = gql`
  mutation importIdentity($type: String!, $secret: String!) {
    importIdentity(type: $type, secret: $secret) {
      did
    }
  }
`

export const CREATE_IDENTITY = gql`
  mutation createIdentity($type: String!) {
    createIdentity(type: $type) {
      did
    }
  }
`

export const DELETE_IDENTITY = gql`
  mutation deleteIdentity($type: String!, $did: String!) {
    deleteIdentity(type: $type, did: $did)
  }
`

export const NEW_MESSAGE = gql`
  mutation handleMessage(
    $raw: String!
    $meta: [MetaDataInput]
    $save: Boolean
  ) {
    handleMessage(raw: $raw, meta: $meta, save: $save) {
      type
      credentials {
        hash
        raw
        credentialSubject
        expirationDate
        issuer {
          did
        }
        subject {
          did
        }
        claims {
          isObj
          type
          value
        }
      }
    }
  }
`

export const NEW_CREDENTIAL = gql`
  mutation handleMessage(
    $raw: String!
    $meta: [MetaDataInput]
    $save: Boolean
  ) {
    handleMessage(raw: $raw, meta: $meta, save: $save) {
      type
      credentials {
        id
        raw
        issuer {
          did
        }
        subject {
          did
        }
        claims {
          did
        }
      }
    }
  }
`

export const GET_MANAGED_IDENTITIES = gql`
  query managedIdentities {
    managedIdentities {
      did
      isSelected
      shortId: shortDid
      shortDid
      profileImage: latestClaimValue(type: "profileImage")
    }
  }
`

export const GET_MESSAGE = gql`
  query GetMessage($hash: String!) {
    message(hash: $hash) {
      jwt
    }
  }
`

export const GET_MESSAGE_SDR = gql`
  fragment ShortProfile on Identity {
    did
    shortId: shortDid
    shortDid
    firstName: latestClaimValue(type: "firstName")
    lastName: latestClaimValue(type: "lastName")
    profileImage: latestClaimValue(type: "profileImage")
  }
  query GetMessageSDR($id: ID!, $selectedIdentity: ID!) {
    message(id: $id) {
      threadId
      id
      replyTo
      replyUrl
      from {
        did
        shortId: shortDid
      }
      sdr(id: $selectedIdentity) {
        claimType
        claimValue
        credentialType
        essential
        reason
        issuers {
          did {
            did
          }
        }
        credentials {
          hash
          raw
          jwt: raw
          issuanceDate
          expirationDate
          credentialSubject
          iss: issuer {
            ...ShortProfile
          }
          sub: subject {
            ...ShortProfile
          }
          fields: claims {
            type
            value
            isObj
          }
        }
      }
    }
  }
`
export const ALL_MESSAGES = gql`
  fragment ShortProfile on Identity {
    did
    shortId: shortDid
    name: latestClaimValue(type: "name")
    firstName: latestClaimValue(type: "firstName")
    lastName: latestClaimValue(type: "lastName")
    profileImage: latestClaimValue(type: "profileImage")
  }
  query AllMessages {
    messages(input: { order: [{ column: saveDate, direction: DESC }] }) {
      id
      saveDate
      updateDate
      createdAt
      expiresAt
      threadId
      type
      raw
      data
      replyTo
      replyUrl
      viewer {
        ...ShortProfile
      }
      presentations {
        hash
        raw
      }
      credentials {
        hash
        raw
        issuer {
          ...ShortProfile
        }
        subject {
          ...ShortProfile
        }
        issuanceDate
        expirationDate
        credentialSubject
        claims {
          type
          value
          isObj
        }
      }
      from {
        ...ShortProfile
      }
      to {
        ...ShortProfile
      }
      metaData
    }
  }
`

export const SIGN_VP = gql`
  mutation signPresentationJwt($data: SignPresentationInput!, $save: Boolean) {
    signPresentationJwt(data: $data, save: $save) {
      raw
    }
  }
`

export const SIGN_VC_MUTATION = gql`
  mutation sign($data: SignCredentialInput!, $save: Boolean) {
    signCredentialJwt(data: $data, save: $save) {
      raw
    }
  }
`

export const SIGN_SDR_MUTATION = gql`
  mutation signSdrJwt($data: SDRInput!) {
    signSdrJwt(data: $data)
  }
`

export const SEND_JWT_MUTATION = gql`
  mutation send($from: String!, $to: String!, $jwt: String!) {
    actionSendJwt(from: $from, to: $to, jwt: $jwt) {
      id
    }
  }
`

export const SEND_DIDCOMM_MUTATION = gql`
  mutation sendMessageDidCommAlpha1(
    $data: SendMessageDidCommAlpha1Input!
    $url: String
  ) {
    sendMessageDidCommAlpha1(data: $data, url: $url, save: true) {
      id
    }
  }
`
