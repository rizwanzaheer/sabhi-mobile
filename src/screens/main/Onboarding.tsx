import * as React from 'react'
import { useTranslation } from 'react-i18next'
import { Screen, Container, Button, Constants, Text } from '@kancha/kancha-ui'
import { NavigationStackProp } from 'react-navigation-stack'

type Props = {
  navigation: NavigationStackProp
}

const Onboarding: React.FC<Props> = ({ navigation }) => {
  const { t } = useTranslation()
  return (
    <Screen background={'primary'}>
      <Container alignItems={'center'} flex={1} padding>
        <Container marginTop={30}>
          <Text type={Constants.TextTypes.H2} bold>
            Get started!
          </Text>
        </Container>
        {/* <Container marginTop={30}>
          <Text type={Constants.TextTypes.Body} textAlign={'center'}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </Text>
        </Container>
        <Container w={300} marginTop>
          <Button
            fullWidth
            type={Constants.BrandOptions.Primary}
            block={Constants.ButtonBlocks.Outlined}
            buttonText={t('Restore seed')}
            onPress={() => navigation.navigate('Restore')}
          />
        </Container> */}
        <Container marginTop={30}>
          <Text type={Constants.TextTypes.Body} textAlign={'center'}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </Text>
        </Container>
        <Container w={300} marginTop={30}>
          <Button
            fullWidth
            type={Constants.BrandOptions.Primary}
            block={Constants.ButtonBlocks.Filled}
            buttonText={t('Create New Identity')}
            onPress={() => navigation.navigate('CreatingWallet')}
          />
        </Container>
      </Container>
    </Screen>
  )
}

export default Onboarding
