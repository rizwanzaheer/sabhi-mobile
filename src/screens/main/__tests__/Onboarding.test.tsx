import 'react-native'
import React from 'react'
import { render } from 'react-native-testing-library'
import { MockedProvider } from '@apollo/react-testing'
import Onboarding from '../Onboarding'
import { GET_MANAGED_IDENTITIES } from '../../../lib/graphql/queries'

jest.useFakeTimers()
jest.runAllTimers()

const mockDidItem = {
  did: 'did:ethr:123456',
  shortId: 'did:ethr:12..3',
  isSelected: true,
}

const mocks = [
  {
    request: {
      query: GET_MANAGED_IDENTITIES,
      variables: {},
    },
    result: {
      data: {
        managedIdentities: [mockDidItem],
      },
    },
  },
]

describe('Onboarding', () => {
  test('renders the intermediate screen if dids are present', async () => {
    // jest.useFakeTimers()
    // @ts-ignore
    const onboarding = <Onboarding navigation={jest.fn()} />
    const tree = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        {onboarding}
      </MockedProvider>,
    ).toJSON()

    // jest.runAllTimers()

    expect(tree).toMatchSnapshot()
  })
})
