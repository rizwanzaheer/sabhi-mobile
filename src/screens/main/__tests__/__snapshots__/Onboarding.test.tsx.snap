// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding renders the intermediate screen if dids are present 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "alignItems": undefined,
      "backgroundColor": "#FFFFFF",
      "borderRadius": undefined,
      "flex": 1,
      "flexDirection": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "margin": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "padding": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "width": undefined,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": 1,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": 30,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "flexWrap": "wrap",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Get started!
      </Text>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": 30,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "flexWrap": "wrap",
            "fontSize": 16,
            "lineHeight": 22,
            "textAlign": "center",
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </Text>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": 30,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": 300,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0078ff",
            "borderRadius": 8,
            "padding": 15,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": "row",
              "height": undefined,
              "justifyContent": "center",
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": "center",
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "flexWrap": "wrap",
                  "fontSize": 18,
                }
              }
            >
              Create New Identity
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
